envoy/ratelimit/ratelimit.h

```diff
@@ -78,11 +70,22 @@ struct LocalDescriptor {
   }
 
   /**
-   * Local descriptor map.
+   * Descriptor map.
    */
-  template <class V> using Map = absl::flat_hash_map<LocalDescriptor, V, Hash, Equal>;
+  template <class V> using Map = absl::flat_hash_map<DescriptorBase, V, Hash, Equal>;
 };
 
+/**
+ * A single rate limit request descriptor. See ratelimit.proto.
+ * This is generated from the request based on the configured rate limit actions.
+ */
+struct Descriptor : public DescriptorBase {
+  absl::optional<RateLimitOverride> limit_ = absl::nullopt;
+  absl::optional<uint32_t> hits_addend_ = absl::nullopt;
+};
+
+using LocalDescriptor = DescriptorBase;
```

Comment:
In the previous implementation, the class `LocalDescriptor` has two usages:
1. as the key of rate limit rules that every descriptor is related to a token bucket.
2. as the output of the descriptor populating of local rate limt filter. It will be used to find a matched rate limit rule.

But now, to support per descriptor `hits_addend`, we need a new class to represent the output of the descriptor populating. Finally, we choose to enhance and re-use the `Descriptor` class. (The `Descriptor` class is used as the output of global rate limit filter. Re-using it could also simplify future development when we want to add similar support for global rate limit filter in the future.)